packages/service-library/src/web-api/web-controller.ts

```diff
@@ -3,31 +3,44 @@
 
 import { Context } from '@azure/functions';
 import { System } from 'common';
-import { inject, injectable } from 'inversify';
+import { injectable, Container, unmanaged } from 'inversify';
 import { ContextAwareLogger } from 'logger';
 import { WebRequestValidator } from './web-request-validator';
+import { WebControllerAuth } from './web-controller-auth';
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
+export interface WebController {
+    aclName?: string;
+}
+
 @injectable()
 export abstract class WebController {
     public context: Context;
 
+    public container: Container;
+
     constructor(
-        @inject(ContextAwareLogger) protected readonly logger: ContextAwareLogger,
-        protected readonly requestValidator: WebRequestValidator,
+        @unmanaged() protected readonly logger: ContextAwareLogger,
+        @unmanaged() protected readonly requestValidator: WebRequestValidator,
     ) {}
 
     public abstract readonly apiVersion: string;
 
     public abstract readonly apiName: string;
 
-    public async invoke(requestContext: Context, ...args: any[]): Promise<unknown> {
+    public async invoke(requestContext: Context, container: Container, ...args: any[]): Promise<unknown> {
         this.context = requestContext;
+        this.container = container;
 
         try {
             this.logger.setCommonProperties(this.getBaseTelemetryProperties());
 
+            const authorized = await this.authorize();
+            if (!authorized) {
+                return;
```

Comment:
Should we return an error code in this case?

Reply:
HTTP 401 set by `WebControllerAuth.authorize()` that accepts request context